Update README on howto deploy custom plans (#1)
rancher-sandbox/elemental-example-plan: README.md

Plans have a `run.sh` file that is executed when a plan is run by `system-agent`.

To build the plan, checkout this repository 

```bash
git clone https://github.com/rancher-sandbox/elemental-example-plan
cd elemental-example-plan
```

edit `files/run.sh`, build container image with it:

```bash
docker build -t elemental-example-plan .
```

and then push the image to a registry, as it should be accessible remotely by nodes.

Plans can be either specified remotely via a management cluster, or locally by writing plan files in `/var/lib/elemental/agent/plans` inside a node host folder, which is running the `rancher-system-agent`.

> Note: Elemental configures the `/var/lib/elemental` path over the standard `/var/lib/rancher` path.

Reviewer: This is true only till the cluster provisioning is triggered on: at that point, a new rancher-system-agent is downloaded and will take over the old instance, moving the configuration files to /var/lib/rancher. The plan for the (hopefully near)  future is to just have an instance and keep the same config for the whole host lifetime... 
so, if you wan to run a local plan *after* cluster provisioning, you need to switch to the `/var/lib/rancher` path.

         group: 0
```

You can create a local plan via a [MachineRegistration CRD](https://github.com/rancher/elemental-operator#machineregistration) , used to register a node in Rancher Manager

```yaml
apiVersion: elemental.cattle.io/v1beta1
kind: MachineRegistration
metadata:
  name: test-nodes
  namespace: fleet-default
spec:
  config:
    cloud-config:
      users:
      - name: root
        passwd: linux
      write_files:
      - path: /var/lib/elemental/agent/plans/test.plan
        permissions: "0600"
        content: |

>           {"instructions":[{"name":"test","image":"ghcr.io/rancher-sandbox/elemental-example-plan:8h"}]}

Reviewer 2: awesome!

         group: 0
```

> When you want to register node in Rancher Manager and create local plan from there:

Reviewer: I would link this to the elemental-operator readme... something like:
```suggestion
You can create a local plan via a [MachineRegistration CRD](https://github.com/rancher/elemental-operator#machineregistration) , used to register a node in Rancher Manager
```
but it is just a nit, feel free to ignore this!

```yaml
apiVersion: elemental.cattle.io/v1beta1
kind: MachineRegistration
metadata:
  name: test-nodes
  namespace: fleet-default
spec:
  config:
    cloud-config:
      users:
      - name: root
        passwd: linux
      write_files:
      - path: /var/lib/elemental/agent/plans/test.plan
        permissions: "0600"
        content: |
          {"instructions":[{"name":"test","image":"ghcr.io/rancher-sandbox/elemental-example-plan:8h"}]}
    elemental:
      install:

>         automatic: true

Reviewer: Automatic was ignored and is no more in the elemental-operator code
```suggestion

```
